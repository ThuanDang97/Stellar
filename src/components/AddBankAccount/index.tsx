import React, { memo, FC } from 'react'
import Image from 'next/image'
import { useTheme } from 'styled-components'

// Components
import Button from '@components/Button'
import {
  BoxShadowStyled,
  ViewStyled,
  TextStyled,
} from '@components/styled-components'
import CheckingBankItem from '@components/CheckingBankItem'
import { WrapperIconStyled } from '@components/Input/InputStyled'

// Props type
import { AddBankAccountProps } from '@self-types/components/AddBankAccount.props'

const AddBankAccount: FC<AddBankAccountProps> = ({
  handleSubmitContinue,
  bankNumbers = [],
  handleAddBankAccount,
}) => {
  const theme = useTheme()

  const isDisableAddBank = bankNumbers.length >= 10
  const isDisableButton = bankNumbers.length === 0

  return (
    <BoxShadowStyled
      data-testid="addBankAccount"
      shadowType="small"
      padding={theme.metrics.dimensions.xmd}
      textAlign="center"
      flexDirection="column"
      alignItems="initial"
      flexWrap="wrap"
    >
      {bankNumbers.map((item) => (
        <CheckingBankItem
          mTop={theme.metrics.dimensions.xs}
          key={`bank-${item.id}`}
          bankNumber={item.bankNumber}
        />
      ))}

      <CheckingBankItem
        mTop={theme.metrics.dimensions.base}
        isDisable={isDisableAddBank}
        handleAddBankAccount={handleAddBankAccount}
      />

      <ViewStyled mTop={theme.metrics.dimensions.xs} display="flex">
        <WrapperIconStyled mRight={theme.metrics.dimensions.xs}>
          <Image src="/icons/icon-lock.svg" alt="icon" layout="fill" />
        </WrapperIconStyled>

        <TextStyled
          color={theme.colors.black}
          fontSize={theme.typography.fontSize.base}
          lineHeight={theme.typography.lineHeight.md}
          fontFamily="AdobeCleanLight"
          textAlign="left"
          pTop={theme.metrics.dimensions.tiny}
          fontWeight={theme.typography.fontWeight.small}
        >
          Bank-level Security
        </TextStyled>
      </ViewStyled>

      <Button
        {...(!isDisableButton && { cursor: 'pointer' })}
        title="Continue"
        onClick={handleSubmitContinue}
        disabled={isDisableButton}
        mTop={theme.metrics.dimensions.xxxl}
        btnWidth={theme.metrics.width.full}
      />
    </BoxShadowStyled>
  )
}

export default memo(AddBankAccount)
