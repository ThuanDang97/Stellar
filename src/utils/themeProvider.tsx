import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
