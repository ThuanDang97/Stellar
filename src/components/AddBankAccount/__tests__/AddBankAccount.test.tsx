// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import AddBankAccount from '@components/AddBankAccount/index'

describe('AddBankAccount Component', () => {
  const component = <AddBankAccount bankNumbers={[]} />

  it('renders ForgotPassword component', () => {
    const { container } = renderWithTheme(component)
    expect(container).toMatchSnapshot()
  })
})
