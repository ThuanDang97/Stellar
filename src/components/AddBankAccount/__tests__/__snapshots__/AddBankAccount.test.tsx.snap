// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBankAccount Component renders ForgotPassword component 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 clNLEZ dbZetz"
    data-testid="addBankAccount"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 fTba-Dr iVIPWO"
      cursor="pointer"
      data-testid="addBankNumbers"
      display="flex"
      width="100%"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="bank-icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        class="TextStyled-sc-9xia41-0 haRzPq"
        width="160"
      >
        Add Checking Account
      </span>
      <div
        class="ViewStyled-sc-rqe5sa-0 ewgISW"
        height="20"
        width="20"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 kqwzQV"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 bdINal kHphsF"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <span
        class="TextStyled-sc-9xia41-0 gNBMLb"
        color="#000"
        font-family="AdobeCleanLight"
        font-size="14"
        font-weight="300"
      >
        Bank-level Security
      </span>
    </div>
    <button
      class="ButtonStyled-sc-1ys9v52-0 jujRLW"
      data-testid="button"
      disabled=""
    >
      Continue
    </button>
  </div>
</div>
`;
